### PR TITLE
Set the target payload for watch inventory collector

### PR DESCRIPTION
**Pull request by agrare**
Set the target payload explicitly and clear it when the refresh is finished.  The post_refresh_ems_cleanup method is called at least from here https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh/refreshers/ems_refresher_mixin.rb#L41-L43 so it will be called even if refresh throws an exception.

Related to: https://github.com/ManageIQ/manageiq/pull/16413
                  https://github.com/ManageIQ/manageiq-providers-openshift/pull/69

### CODE REVIEW

File: app/models/manageiq/providers/kubernetes/container_manager/target_collection_mixin.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def pods_references(pods)
   end
 
   def pod_list
-    target.targets.map { |target| JSON.parse(target.options[:payload], :object_class => OpenStruct) }
+    target.targets.map { |target| JSON.parse(target.payload, :object_class => OpenStruct) }
```
Comment (Ladas) [inline app/models/manageiq/providers/kubernetes/container_manager/target_collection_mixin.rb]:
```ruby
target.targets.map { |target| JSON.parse(target.payload, :object_class => OpenStruct) if target.payload }.compact
```

File: spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb [hunk 2]
```diff
@@ -62,10 +62,7 @@ def queue_target!(watch_data_path)
       :manager     => @ems,
       :association => :container_groups,
       :manager_ref => ems_ref,
-      :options     => {
-        :payload => notice['object'].to_json,
-      },
-    )
+    ).tap { |t| t.payload = notice['object'].to_json }
```
Comment (cben) [inline spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb]: [cosmetic, your call] don't need .tap, can simply `target.payload = ...` on separate line.

Comment (agrare) [reply spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb]: I was keeping this the same as the call in https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/164/files#diff-03ceb0538d4c3023378bbb63123f770dR72 but that is returning the result in that line, you're right no need for the tap here

**Comment by agrare** [comment]
> are PR title & description still accurate? I'm not sure where clearing is now done...

Not anymore, at first I was clearing the payload in a post_ems_refresh handler but now we have an association between the queue item and the binary blob so we don't need to explicitly clear the payload here.  I'll update the description

File: spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb (hunk 2 above)
Comment (cben) [reply spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb]: sorry, that was 100% silly bikeshedding compared to the non-trivial content of these PRs, doesnt matter either way :bike: :house:

Comment (agrare) [reply spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb]: You did say it was cosmetic and I agree this is better so no need to apologize :)
We should have @Ladas update his PR (he did the same thing) https://github.com/ManageIQ/manageiq-providers-openshift/pull/68/files#diff-d6b73355e83b934cb272f17e43c007e3R61 for consistency

Comment (Ladas) [reply spec/models/manageiq/providers/kubernetes/container_manager/targeted_refresh/targeted_refresh_spec.rb]: updated, new PR https://github.com/ManageIQ/manageiq-providers-openshift/pull/69

**Comment by moolitayer** [comment]
@kbrock please review

**Comment by moolitayer** [comment]
@kbrock @cben please take a look

File: app/models/manageiq/providers/kubernetes/container_manager/inventory_collector_mixin.rb [hunk 3]
```diff
@@ -63,14 +63,13 @@ def parse_target_from_notice(notice)
...
       :association => :container_groups,
-      :manager_ref => ems_ref,
-      :options     => {
-        :payload => object.to_json,
-      }
-    )
+      :manager_ref => ems_ref
+    }
+
+    ManagerRefresh::Target.new(target_opts).tap { |target| target.payload = object.to_json }
```
Comment (cben) [inline app/models/manageiq/providers/kubernetes/container_manager/inventory_collector_mixin.rb]: What took me a while to grok about this code is that for 1 conceptual goal of "queue this via DB" is split between 2 *different looking* code channels:
- a BinaryBlob is written to DB immediately in `Target#payload=`.
- the Target, holding among others blob id, is an in-memory ruby object — not even an ActiveRecord — whose content gets written into DB later, inside `EmsRefresh.queue_refresh` via `Target#dump`.

[not opposing and don't have any better suggestion, just hoping re-arch obsoletes this by giving us a more scalable queue...]

**Comment by cben** [comment]
Ah, I see, https://github.com/ManageIQ/manageiq/pull/16413 dependency is still unmerged.

**Pull request closed without merging**